### DISCUSSION TITLE
AttributeError: 'Tensor' object has no attribute 'ref'

### DISCUSSION THREAD

**Discussion by saini056** [lululxvi/deepxde#1186] [General]
I am trying to use dde.grad.jacobian and dde.grad.hessian to calculate gradients for my one of my case and it is giving me this error:

File "/home/tutorial/saini056/.local/lib/python3.9/site-packages/deepxde/gradients.py", line 146, in __call__
    key = (ys.ref(), xs.ref())
AttributeError: 'Tensor' object has no attribute 'ref'

I am using Tensorflow==2.11.0
Can you please help with this? Thank you.

**Comment by lululxvi** [comment]
Read the examples.
